testutils/src/main/java/com/microsoft/identity/internal/testutils/TestUtils.java

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.internal.testutils;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.test.core.app.ApplicationProvider;
+
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+import com.microsoft.identity.common.internal.dto.CredentialType;
+
+import org.mockito.Mockito;
+
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import static com.microsoft.identity.common.internal.cache.CacheKeyValueDelegate.CACHE_VALUE_SEPARATOR;
+
+public class TestUtils {
+
+    private static String getCacheKeyForAccessToken(Map<String, ?> cacheValues) {
+        for (Map.Entry<String, ?> cacheValue : cacheValues.entrySet()) {
+            final String cacheKey = cacheValue.getKey();
+            if (isAccessToken(cacheKey)) {
+                return cacheKey;
+            }
+        }
+
+        return null;
+    }
+
+
+    /**
+     * Inspects the supplied cache key to determine the target CredentialType.
+     *
+     * @param cacheKey The cache key to inspect.
+     * @return The CredentialType or null if a proper type cannot be resolved.
+     */
+    @Nullable
+    private static CredentialType getCredentialTypeForCredentialCacheKey(@NonNull final String cacheKey) {
```

Comment:
If you want to refactor the original implementation so it doesn't get duplicated here, that seems preferable

Reply:
Okay, I'll change the original implementation to be `public static` so we can call it in our tests. Based on the nature of the method, I think that is fine to do as it does not seems necessary for this method to be an instance method or be private either.

Reply:
Addressed